cmd/minishift/cmd/start.go

```diff
@@ -350,40 +355,45 @@ func startHost(libMachineClient *libmachine.Client) *host.Host {
-		DockerEngineOpt:     getSlice(configCmd.DockerEngineOpt.Name),
-		InsecureRegistry:    determineInsecureRegistry(configCmd.InsecureRegistry.Name),
-		RegistryMirror:      getSlice(configCmd.RegistryMirror.Name),
-		HostOnlyCIDR:        viper.GetString(configCmd.HostOnlyCIDR.Name),
-		HypervVirtualSwitch: viper.GetString(configCmd.HypervVirtualSwitch.Name),
-		ShellProxyEnv:       shellProxyEnv,
+		MinikubeISO:           determineIsoUrl(viper.GetString(configCmd.ISOUrl.Name)),
+		ISOCacheDir:           state.InstanceDirs.IsoCache,
+		Memory:                calculateMemorySize(viper.GetString(configCmd.Memory.Name)),
+		CPUs:                  viper.GetInt(configCmd.CPUs.Name),
+		DiskSize:              calculateDiskSize(viper.GetString(configCmd.DiskSize.Name)),
+		VMDriver:              viper.GetString(configCmd.VmDriver.Name),
+		DockerEnv:             append(dockerEnv, getSlice(configCmd.DockerEnv.Name)...),
+		DockerEngineOpt:       getSlice(configCmd.DockerEngineOpt.Name),
+		InsecureRegistry:      determineInsecureRegistry(configCmd.InsecureRegistry.Name),
+		RegistryMirror:        getSlice(configCmd.RegistryMirror.Name),
+		HostOnlyCIDR:          viper.GetString(configCmd.HostOnlyCIDR.Name),
+		HypervVirtualSwitch:   viper.GetString(configCmd.HypervVirtualSwitch.Name),
+		ShellProxyEnv:         shellProxyEnv,
+		RemoteIPAddress:       viper.GetString(configCmd.RemoteIPAddress.Name),
+		RemoteSSHUser:         viper.GetString(configCmd.RemoteSSHUser.Name),
+		SSHKeyToConnectRemote: viper.GetString(configCmd.SSHKeyToConnectRemote.Name),
 	}
 	minishiftConfig.InstanceConfig.VMDriver = machineConfig.VMDriver
 	minishiftConfig.InstanceConfig.Write()
 
 	fmt.Printf(" using '%s' hypervisor ...\n", machineConfig.VMDriver)
 	var hostVm *host.Host
 
-	// configuration with these settings only happen on create
-	isRestart := cmdUtil.VMExists(libMachineClient, constants.MachineName)
-	if !isRestart {
-		fmt.Println("-- Minishift VM will be configured with ...")
-		fmt.Println("   Memory:   ", units.HumanSize(float64((machineConfig.Memory/units.KiB)*units.GB)))
-		fmt.Println("   vCPUs :   ", machineConfig.CPUs)
-		fmt.Println("   Disk size:", units.HumanSize(float64(machineConfig.DiskSize*units.MB)))
-	}
+	if machineConfig.VMDriver != genericDriver {
+		// configuration with these settings only happen on create
+		isRestart := cmdUtil.VMExists(libMachineClient, constants.MachineName)
+		if !isRestart {
+			fmt.Println("-- Minishift VM will be configured with ...")
+			fmt.Println("   Memory:   ", units.HumanSize(float64((machineConfig.Memory/units.KiB)*units.GB)))
+			fmt.Println("   vCPUs :   ", machineConfig.CPUs)
+			fmt.Println("   Disk size:", units.HumanSize(float64(machineConfig.DiskSize*units.MB)))
+		}
 
-	configureNetworkSettings()
+		configureNetworkSettings()
 
-	cacheMinishiftISO(machineConfig)
+		cacheMinishiftISO(machineConfig)
 
-	fmt.Print("-- Starting Minishift VM ...")
+		fmt.Print("-- Starting Minishift VM ...")
+	}
```

Comment:
@praveenkumar May be in future we can , inform the configuration of the remote host to the user when generic driver is used. It would be a relatively easy task.